fixes(player-controls): bug fix for player-controls and gestures.

himanshu-chicmic opened the pull request:
## Pull Request
### Description
This PR addresses the following issues:
- When video seeks with double tap gestures, the seek button hides when controls are visible or appears when controls are not visible.
- Dispatch queue use, the controls are supposed to hide after 5 seconds interval, but when a tap event is received on videoContainer the process continues.
- When controls are locked, other controls get's visible with videoContainer tap.
- Video seek buffer time uses int and doesn't seek time or video precisely.
- When video ends seek back or forward button gets visible on videoContainer tap event.
- Touch points for seek back or forward are not properly set.
- Prevent video to seek or zoom when controls are locked.

### Changes Made
- Seek buttons uses controlsHidden var for visibility.
- For hide controls after 5 seconds, dispatch queue is not cancelled when user taps before the hiding time and the queue is reinitialized when controls are visible again.
- When video controls are locked only lock icon is shown
- Changed video seek time from Int to double for more precision.
- When a video ends playing, seek and forward buttons are not shown.
- double touch points are set based on the midX of the view.
- when controls locked, zoom and seek gestures are also disabled.

### Related Issues


### Testing Done


### Checklist
- [ ] Manual testing has been performed and passed.
- [ ] Documentation has been updated (if applicable).
- [ ] The branch is up to date with the target branch (e.g., `main` or `master` or `development` or `staging`).

### Additional Notes

Commit:
feat(controls): add play and pause button

Commit:
feat: add mute button

Commit:
new commit

Vikram-Chicmic committed:
feat(custom-slider): add custom-slider

himanshu-chicmic committed:
feat(zoom-in-out): add zoom in and out feature to video player.

Commit:
feat(controls): add media player controls

Commit:
update: update the buffer time

Commit:
- delete(Control buttons folder)

Vikram-Chicmic committed:
feat(tap-and-drag-to-seek): Add tap and drag feature to seek or jump in video

Vikram-Chicmic committed:
fix(distortion-while-seeking): resolve the distortion of seek bar while dragging

himanshu-chicmic committed:
merge(control-buttons): merge control-buttons branch into zoom-in-out branch.

himanshu-chicmic committed:
merge(seek-bar-customization): merge seek-bar-customization branch into zoom-in-out

Vikram-Chicmic committed:
feature(add-time-stamps): Added time labels or time stamps on seek-bar

himanshu-chicmic committed:
feat(player-controls): add close, play, pause, replay, seek forward, seek backward and mute button.

himanshu-chicmic committed:
feat(slider-time-view): add slider and time view to video player with code optimizations in "SliderTimeLabelView.swift" and "TimeLabels.swift".

himanshu-chicmic committed:
feat(add-controls, functionality): add controls button view for lock video, mute, stop video player when close, update code with optimizations.

himanshu-chicmic committed:
refactor(code-optimizations): update code with swiftlint, changed default enum values in IconSize and ButtonSize.

himanshu-chicmic committed:
feat(controls-and-tap-gestures): add tap gestures for seek controls, animation for seek buttons, add auto hide controls after a time interval.

himanshu-chicmic committed:
fix(player-controls): fixed show/hide function for buttons, updated code for touch gestures to change the visibility of player controls. when view is locked, do not show any other controls, when video is zoomed do not display close button but other control are still visible. updated double tap to seek by getting the touch point based on the midX position of the view.

himanshu-chicmic committed:
feat(ib-designables): converted mute button, lock control to ib designables components for use through storyboard.

SimranGarg24 commented on CustomAVPlayer/Enums/ControlButtons.swift:
```diff
@@ -0,0 +1,69 @@
+    case pauseRectangleFill     = "pause.rectangle.fill"
+}
+
+public enum ReplayButtonImage: String {
```
what is the need of enum for single case

SimranGarg24 commented on CustomAVPlayer/Enums/ControlButtons.swift:
```diff
@@ -0,0 +1,69 @@
+public enum ForwardButtonImage: String {
+    case forwardButton          = "goforward.5"
+}
+
```
These both enums can be combined into one

SimranGarg24 commented on CustomAVPlayer/Controls/ForwardBackwardButton.swift:
```diff
@@ -0,0 +1,105 @@
+        self.heightAnchor.constraint(equalToConstant: size.rawValue).isActive = true
+        self.widthAnchor.constraint(equalToConstant: size.rawValue).isActive = true
+    }
+    
```
Avoid Use of force unwrapping and add comments on this method

SimranGarg24 commented on CustomAVPlayer/Views/CustomSlider.swift:
```diff
@@ -0,0 +1,152 @@
+        self.isUserInteractionEnabled = interactionEnabled
+    }
+    
+    public override func beginTracking(_ touch: UITouch, with event: UIEvent?) -> Bool {
```
Add comment

SimranGarg24 commented on CustomAVPlayer/Views/SliderTimeLabelView.swift:
```diff
@@ -0,0 +1,79 @@
+        self.alignment = .fill
+        self.distribution = .fill
+    }
+
```
No need to remove views then add, they can be swapped even without removing them

himanshu-chicmic committed:
feat(change-player-controls): Change forward, back, play, pause, replay buttons to IB Designabels.
Add these controls to VideoPlayerView.xib
Remove av player instance from custom button files.

himanshu-chicmic committed:
refactor(custom-buttons): created a single common method for set up of custom button view, made seek buttons (forward and backward) re-appear when video is replayed.

himanshu-chicmic committed:
fix(designables): changed UIViewController to UIView for user as a subview inside ViewController, made it IBDesignable to add through storyboard. Created all related IBProperties to button controls in main file for change through Interface Builder.

himanshu-chicmic committed:
refactor(code-updation): remove setButtonImage() function

himanshu-chicmic committed:
feat(update-designables): changed all controls with ibproperties, separated code for better readability.

Vikram-Chicmic committed:
fixes(custom-slider): slider emphasis on drag gesture and synchronization of slider with buttons

himanshu-chicmic committed:
fixes(gestures-controls): add long tap gesture for pause video when controls are disabled, added controls disabled check for show/hide icons.
